File: 0_Project_CURRENT/Assets/scripts/LightShooter.js
var lightPF: Transform;
var Zspeed: float;
var maxZ: float;
private var counter: float=0.0;
private var timeInc: float=0.05;
private var chance: float=0.2;
var lightPPs: Transform[];
var camTrans: Transform;
var backlights: Light[];
var worldNum: int;
var rando: float;

function Start() {
	var camGO: GameObject = GameObject.Find("Main Camera");
	camTrans=camGO.transform;
	var which: String = "warpLight"+worldNum;
	backlights[0].GetComponent.<Animation>().Play(which);
	//backlights[1].color=liteColors[worldNum];
}

function Update () {
	for (var i: int=0; i<2; i++) {
		//if (lightPPs[i]) {
			//print (i);
			var lite = lightPPs[i];
			lite.localPosition.z+=Zspeed*Time.deltaTime*10;
			if (lite.localPosition.z>maxZ) {
				respawnLight(lite.transform);
			}
		//}
	}

	backlights[0].transform.localEulerAngles.y=180+camTrans.position.x*-36;
	backlights[1].transform.localEulerAngles.y=180+camTrans.position.x*-8;
	
}

function respawnLight(obj: Transform) {
	//var rando: float = Random.Range(-0.5, 3.64);
	//print (rando);
	//var x = Mathf.Cos(rando);
	//var y = Mathf.Sin(rando);
	var x = Random.Range(-2.0, 2.0)+camTrans.position.x;
	var y = 5;
	var z = Random.Range(-20.0, -12.0);
	rando = Random.Range(0.25, 1.5);
	var pos: Vector3 = Vector3(x, y, z)*rando;

	obj.localPosition=pos;
	
	//var lite: Light = obj.GetComponent(Light);
	//rando = Random.Range(0.0, 1.0);
	//var liteColor: Color = Color(rando, Mathf.Clamp((rando*1.6+0.2), 0.0, 1.0), 1, 1);
	//lite.color=liteColor;
}

File: 0_Project_CURRENT/Assets/scripts/LightShooter.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LightShooter : MonoBehaviour
{

	public Transform lightPF;
	public float Zspeed;
	public float maxZ;

	Transform camTrans;
	public Transform[] lightPPs;
	public Light[] backlights;
	public int worldNum;

	void Start ()
	{
		GameObject camGO = GameObject.Find ("Main Camera");
		camTrans = camGO.transform;
		string which = "warpLight" + worldNum;
		backlights [0].GetComponent<Animation> ().Play (which);
		//backlights[1].color=liteColors[worldNum];
	}

	void Update ()
	{
		foreach (Transform lite in lightPPs) {
			lite.localPosition += new Vector3 (0, 0, Zspeed * Time.deltaTime * 10);
			if (lite.localPosition.z > maxZ) {
				respawnLight (lite);
			}
		}

		backlights [0].transform.localEulerAngles = new Vector3 (10, 180 + camTrans.position.x * -36, 0);
		backlights [1].transform.localEulerAngles = new Vector3 (14, 180 + camTrans.position.x * -8, 0);

	}

	void respawnLight (Transform obj)
	{
		float x = Random.Range (-2, 2) + camTrans.position.x;
		float y = 5;
		float z = Random.Range (-20, -12);
		float rando = Random.Range (0.25f, 1.5f);

		obj.localPosition = new Vector3 (x, y, z) * rando;

		//var lite: Light = obj.GetComponent(Light);
		//rando = Random.Range(0.0, 1.0);
		//var liteColor: Color = Color(rando, Mathf.Clamp((rando*1.6+0.2), 0.0, 1.0), 1, 1);
		//lite.color=liteColor;
	}

}
